File: 70. 爬楼梯/70. 爬楼梯 -递归超时.py
class Solution:
    def climbStairs(self, n: int) -> int:
        # 爬到第10层楼的所有方法为（从8楼爬2层+从9楼爬一层，注意8+1+1包含在9+1里面）
        def r(a):
            if a==1:
                return 1
            if a==2:
                return 2
            return r(a-1)+r(a-2)
        
        return r(n)

File: 70. 爬楼梯/70. 爬楼梯-递归超时.go
package main

import "fmt"

func main() {
	fmt.Println(climbStairs(45))
}


// 递归实现，在n=44超时了，本地运行都跑了9秒
func climbStairs(n int) int {
    // 答案是前两个的和
    if n == 1{
        return 1
    }
    if n == 2{
        return  2
    }
    return climbStairs(n-1) + climbStairs(n-2)
}


// 通过找规律发现答案是前两个之和